mapo/envs/tests/test_navigation.py

```diff
@@ -144,21 +133,19 @@ def test_transition(env):
             assert np.allclose(noise_cov, env._noise["cov"], atol=1e-1)
 
 
-@pytest.mark.parametrize("env", get_gym_envs())
 def test_reward(env):
     with env._graph.as_default():
         dummy_action = env.action_space.sample()
-        dummy_next_state = env.observation_space.sample()
+        dummy_state = env.observation_space.sample()
 
-        reward = env._reward(env._end, dummy_action, dummy_next_state)
+        reward = env._reward(dummy_state, dummy_action, env._end)
         assert np.allclose(reward, 0.0)
-        reward = env._reward(env._start, dummy_action, dummy_next_state)
+        reward = env._reward(dummy_state, dummy_action, env._start)
```

Comment:
Imagine the agent arrives at the goal in the second to last time step. Then, even if the agent moves away from the goal in the final step, we would be rewarding that behavior (the transition would have high reward even though it is a bad one).

Reply:
If the agent arrives at the goal (or close by some metric), then there's no further steps as the episode will terminate.